modules/gateway/gateway.go

```diff
@@ -159,18 +145,26 @@ func New(addr string, persistDir string) (g *Gateway, err error) {
 	g.log.Println("INFO: gateway created, started logging")
 
 	// Forward the RPC port, if possible.
-	go g.forwardPort(port)
+	g.threads.Add()
+	go func() {
+		defer g.threads.Done()
+		g.forwardPort(port)
+	}()
 
 	// Learn our external IP.
-	go g.learnHostname(port)
+	g.threads.Add()
+	go func() {
+		defer g.threads.Done()
```

Comment:
it sort-of does, but it introduces a race condition, because you can `Add` right when `Wait` thinks it's finished. With `ThreadGroup`, calls to `Add` during `Stop` return immediately with an error, so there's no race condition.